memory crash  terraform-provider-azurerm_v3.61.0_x5.exe

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.5.1

### AzureRM Provider Version

3.61.0

### Affected Resource(s)/Data Source(s)

data.azurerm_virtual_machine_scale_set

### Terraform Configuration Files

```hcl
data "azurerm_virtual_machine_scale_set" "WriterBuildAgent" {
  name                = var.vmss_build_agents["writer"].name
  resource_group_name = var.vmss_build_agents["writer"].resource_group_name
}
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.61.0_x5.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x5a6a25a]

goroutine 117 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.getVirtualMachineScaleSetVMConnectionInfo({0x832f608, 0xc0023ac2a0}, {0xc002525270, 0x1, 0xd18c2e2800?}, {0xc002394450, 0x12}, {0xc0023a2340, 0x1a}, {0xc0
01b83f96, ...}, ...)
        github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/virtual_machine_scale_set_data_source.go:235 +0x79a
github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.dataSourceVirtualMachineScaleSetRead(0x0?, {0x6e37240?, 0xc001a43400?})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/compute/virtual_machine_scale_set_data_source.go:184 +0xf0a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x832f640?, {0x832f640?, 0xc0013e8b10?}, 0xd?, {0x6e37240?, 0xc001a43400?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc000b6c1c0, {0x832f640, 0xc0013e8b10}, 0xc000299b00, {0x6e37240, 0xc001a43400})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:943 +0x145
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc00187ea38, {0x832f640?, 0xc0013e89f0?}, 0xc001b40840)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1195 +0x38f
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0008f5d60, {0x832f640?, 0xc002c1e180?}, 0xc0005b60a0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:658 +0x403
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x761f380?, 0xc0008f5d60}, {0x832f640, 0xc002c1e180}, 0xc0019d8000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000260000, {0x833faa0, 0xc001b7c000}, 0xc000826000, 0xc001bf6450, 0xc985710, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc000260000, {0x833faa0, 0xc001b7c000}, 0xc000826000, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.61.0_x5.exe plugin crashed!
```


### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

1 terraform plan

### Important Factoids

os windows 11

### References

_No response_

--- a/internal/services/compute/virtual_machine_scale_set_data_source.go
+++ b/internal/services/compute/virtual_machine_scale_set_data_source.go
@@ -3,6 +3,7 @@ package compute
 import (
 	"context"
 	"fmt"
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
 	"strings"
 	"time"
 
@@ -225,14 +226,14 @@ func getVirtualMachineScaleSetVMConnectionInfo(ctx context.Context, networkInter
 						return nil, fmt.Errorf("reading Public IP Address for VM Instance %q for Virtual Machine Scale Set %q (Resource Group %q): %+v", virtualmachineIndex, virtualMachineScaleSetName, resourceGroupName, err)
 					}
 
-					if *nic.Primary && *props.Primary {
+					if pointer.From(nic.Primary) && pointer.From(props.Primary) {
 						primaryPublicAddress = *publicIPAddress.IPAddress
 					}
 					publicIPAddresses = append(publicIPAddresses, *publicIPAddress.IPAddress)
 				}
 
 				if props.PrivateIPAddress != nil {
-					if *nic.Primary && *props.Primary {
+					if pointer.From(nic.Primary) && pointer.From(props.Primary) {
 						primaryPrivateAddress = *props.PrivateIPAddress
 					}
 					privateIPAddresses = append(privateIPAddresses, *props.PrivateIPAddress)

--- a/internal/services/compute/virtual_machine_scale_set_data_source.go
+++ b/internal/services/compute/virtual_machine_scale_set_data_source.go
@@ -3,10 +3,10 @@ package compute
 import (
 	"context"
 	"fmt"
-	"github.com/hashicorp/go-azure-helpers/lang/pointer"
 	"strings"
 	"time"
 
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
 	"github.com/hashicorp/go-azure-helpers/resourcemanager/location"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"

